src/crimson/os/seastore/onode_manager/staged-fltree/fltree_onode_manager.h

```diff
@@ -78,14 +90,35 @@ struct FLTreeOnode final : Onode, Value {
     return *read_payload<onode_layout_t>();
   }
 
-  onode_layout_t &get_mutable_layout(Transaction &t) final {
+  template <typename layout_func_t, typename recorder_func_t>
+  void with_mutable_layout(
+    Transaction &t,
+    layout_func_t &&layout_func,
+    recorder_func_t &&recorder_func) {
     assert(status != status_t::DELETED);
     auto p = prepare_mutate_payload<
       onode_layout_t,
       Recorder>(t);
     status = status_t::MUTATED;
-    return *reinterpret_cast<onode_layout_t*>(p.first.get_write());
-  };
+    layout_func(*reinterpret_cast<onode_layout_t*>(p.first.get_write()));
+    if (p.second) {
+      recorder_func(p.first, *p.second);
+    }
+    status = status_t::STABLE;
+  }
+
+  template <typename layout_func_t>
+  void with_mutable_layout(
+    Transaction &t,
+    layout_func_t &&layout_func) {
+    assert(status != status_t::DELETED);
+    auto p = prepare_mutate_payload<
+      onode_layout_t,
+      Recorder>(t);
+    status = status_t::MUTATED;
+    layout_func(*reinterpret_cast<onode_layout_t*>(p.first.get_write()));
+    populate_recorder(t);
+  }
 
   void populate_recorder(Transaction &t) {
```

Comment:
If UPDATE_ALL can be replaced by CREATE_DEFAULT, probably the special version of `with_mutable_layout()` above, `populate_recorder()`, `record_delta()` can also be dropped.

With the above changes, looks `status_t` only needs 2 states -- ALIVE and DELETED, right?